packages/compass-collection/src/modules/tabs.ts

```diff
@@ -182,56 +183,80 @@ const reducer: Reducer<CollectionTabsState> = (
       tabs: newTabs,
     };
   }
+  if (action.type === CollectionTabsActions.CollectionRenamed) {
+    const { tabs, activeTabIndex } = action;
+
+    const activeTabId = tabs[activeTabIndex]?.id;
+    return {
+      ...state,
+      tabs,
+      activeTabId,
+    };
+  }
   return state;
 };
 
 const subtabChanged = (id: string, name: string) => {
   return { type: CollectionTabsActions.SubtabChanged, id, name };
 };
 
+function createNewTabInternal(
+  collectionMetadata: CollectionMetadata,
+  dispatch: any,
```

Comment:
If you need access to `dispatch`, I'd suggest turning this into a thunk action instead of passing it around without a proper type. Alternatively you can set up the interface of this method in a way that doesn't require you to pass it here, like passing a less specific onSubtabChange callback or something like (or not factoring this subscription in a shared method at all)

Reply:
`dispatch` is typed locally, I just haven't pushed those changes yet